lib/common/db-renderable-content.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+module.exports = dbRenderableContentServiceFactory;
+
+dbRenderableContentServiceFactory.$provide = 'DbRenderableContent';
+dbRenderableContentServiceFactory.$inject = [
+    'FileLoader',
+    'Constants',
+    'Promise',
+    '_',
+    'Services.Waterline',
+    'Logger',
+    'ejs'
+];
+
+/**
+ * dbRenderableContentServiceFactory provides the template-service singleton object.
+ * @private
+ * @param  {FileLoader} FileLoader A FileLoader class for extension.
+ * @param  {configuration} configuration      An instance of the configuration configuration object.
+ * @return {TemplateService}            An instance of the TemplateService.
+ */
+function dbRenderableContentServiceFactory(
+    FileLoader, 
+    Constants, 
+    Promise, 
+    _, 
+    waterline, 
+    Logger, 
+    ejs ) 
+{
+    var logger = Logger.initialize(dbRenderableContentServiceFactory);
+
+    function DbRenderableContentService () {
+        this.loader = new FileLoader();
+    }
+
+    DbRenderableContentService.prototype.load = function load() {
+        var self = this;
+
+        return this.loader.getAll(
+            self.directory
+        ).then(function (templates) {
+            var promises = _.map(templates, function (contents, name) {
+                return self.put(name, contents.toString());
+            });
+
+            return Promise.all(promises);
+        }).catch(function (err) {
+            if (process.env.NODE_ENV !== 'test') {
```

Comment:
@benbp I think it's a copy-pasta from legacy patterns that John set up about not wanting to see his own warnings in unit tests. Don't know if it's relevant any longer, may be exactly the opposite.